### DISCUSSION TITLE
Class "MongoDB\Driver\Manager" not found

### DISCUSSION THREAD

**Discussion by jeanali** [jenssegers/laravel-mongodb#2214] [Q&A] [answered]
- Laravel-mongodb Version: ^3.8
- PHP Version: 8.0
- Laravel: 8

I'm connect to mongodb, it works and seed some data in database and also seed it's done but when I get books from mongodb I got this error:

``` Class "MongoDB\Driver\Manager" not found ```

### My Config:

<img width="639" alt="Screen Shot 2021-03-04 at 4 39 54 PM" src="https://user-images.githubusercontent.com/67483209/109972982-e6c7db80-7d08-11eb-80f4-069972c63f72.png">

- brew services 
<img width="1300" alt="Screen Shot 2021-03-04 at 4 40 16 PM" src="https://user-images.githubusercontent.com/67483209/109973009-efb8ad00-7d08-11eb-9ba4-3684de19aced.png">

 - phpinfo():
<img width="537" alt="Screen Shot 2021-03-04 at 4 41 25 PM" src="https://user-images.githubusercontent.com/67483209/109973020-f34c3400-7d08-11eb-9092-ed351d2113a1.png">

php ini:

```extension="/usr/local/Cellar/php/8.0.2/pecl/20200930/mongodb.so"```

Am I missing something ?

**Comment by Smolevich** [comment]
@jeanali what is the method run php? Php cli or php-fpm? If believe screenshot, you can use php cli. What is the output for command `php -m` and `phpinfo()` in some test script

**Reply by w99910** [reply]
@Smolevich I'm also having this issue. Migrating database is fine but accessing the web got error. Here is php -m command.
I'm using 
- laravel 8.12 
- laravel-mongodb 3.8 
- php  7.4.3
- nginx 
- mongo 4.4.4
I have already add extension extension=mongodb.so in /etc/php/7.4/cli/php.ini .   
![Screenshot from 2021-03-11 22-37-57](https://user-images.githubusercontent.com/65523475/110817568-d42a4500-82ba-11eb-8045-85986a0f5486.png)

**Comment by kriiv** [comment]
Was there a solution found for this?

**Reply by alcaeus** [reply, marked as answer]
Make sure the extension is installed in all SAPIs, not just in CLI (visible when running `php -m`), but also in the FPM SAPI (visible when running `phpinfo()` through a web server). Depending on where your extension is unavailable, check the list of loaded INI files and ensure that one of these files specifies the MongoDB extension.

**Comment by nuryagdym** [comment]
I had this issue with php-fpm, because I installed mongodb extension **after** starting php fpm docker container/service. 
In this case php-fpm needs to be restarted to reload extensions.

One thing that confused me all these commands showed me that extensions are loaded:
php --ini
php -m

But without php-fpm restart, mongodb extension is loaded only to php cli commands. 
Any http requests won't have mongodb extension unless php-fpm is restarted.
